src/internal/coin.ts

```diff
@@ -0,0 +1,32 @@
+import { AptosConfig } from "../api/aptos_config";
+import { U64 } from "../bcs/serializable/move-primitives";
+import { Account, AccountAddress } from "../core";
+import { GenerateTransactionOptions, SingleSignerTransaction } from "../transactions/types";
+import { StructTag, TypeTagStruct } from "../transactions/typeTag/typeTag";
+import { HexInput, AnyNumber, MoveResourceType } from "../types";
+import { APTOS_COIN } from "../utils/const";
+import { generateTransaction } from "./transaction_submission";
+
+export async function transaferCoinTransaction(args: {
```

Comment:
So, we can't call it just `transfer` because everything is unioned together on a single interface?

Reply:
in the future, we will have support for FA, NFT, Object, etc. since they all under the same namespace we can't call each `transfer`. we can always but each related transfer function under the each namespace, but that's just confusing for the user imo.
also, I think transferCoin makes sense, as this function transfers coin